go-controller/pkg/ovn/gress_policy.go

```diff
@@ -156,7 +165,17 @@ func (gp *gressPolicy) addPeerPods(pods ...*v1.Pod) error {
 		if err != nil {
 			return err
 		}
+
+		// Sorting to be sure the same lock sequence is done also in `deletePeerPod`
+		sort.Slice(podIPs, func(i, j int) bool { return podIPs[i].String() < podIPs[j].String() })
```

Comment:
maybe we sort  and lock the whole `ips` list outside of this loop? that would make sure the ordering is correct not only per-pod, but per-gressPolicy
we create a `map[podIP]pod.UID` (this would also make sure we lock every ip only once, the chance that we get same ips for different pods are close to zero, but since it will be a deadlock, I prefer to be safe :) ) then we can sort all keys, and lock in the sorted keys order
wdyt @zeeke ?

Reply:
Makes sense, thanks for pointing out